internal/policy/kubernetes_client_test.go

```diff
@@ -41,33 +43,83 @@ func init() {
 	fakeClient = fake.NewClientBuilder().WithScheme(scheme).WithObjects(&testECP).Build()
 }
 
+const testKubeconfig = `
+apiVersion: v1
+kind: Config
+contexts:
+- context:
+    namespace: test
+  name: test/my-cluster-com:6443/kube:admin
+current-context: test/my-cluster-com:6443/kube:admin
+`
+
 func Test_FetchEnterpriseContractPolicy(t *testing.T) {
-	k := kubernetes{
-		client: fakeClient,
+	testCases := []struct {
+		name           string
+		namespacedName types.NamespacedName
+		ecp            *ecp.EnterpriseContractPolicy
+		kubeconfig     string
+		err            string
+	}{
+		{
+			name:           "fetch-with-name-and-namespace",
+			namespacedName: types.NamespacedName{Name: "ec-policy", Namespace: "test"},
+			ecp:            &testECP,
+			kubeconfig:     "",
+			err:            "",
+		},
+		{
+			name:           "fetch-with-name-only",
+			namespacedName: types.NamespacedName{Name: "ec-policy"},
+			ecp:            &testECP,
+			kubeconfig:     testKubeconfig,
+			err:            "",
+		},
+		{
+			name:           "fetch-with-undetectable-namespace",
+			namespacedName: types.NamespacedName{Name: "ec-policy"},
+			ecp:            nil,
+			kubeconfig:     "",
+			err:            "Unable to determine current namespace: missing current context",
+		},
+		{
+			name:           "fetch-policy-not-found",
+			namespacedName: types.NamespacedName{Name: "ec-policy", Namespace: "missing"},
+			ecp:            nil,
+			kubeconfig:     "",
+			err:            `enterprisecontractpolicies.appstudio.redhat.com "ec-policy" not found`,
```

Comment:
The expected message changed because the previous message was not correct, which highlighted a previous bug in the tests that is now fixed.